library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/DvbSubtitleReader.java

```diff
@@ -56,9 +56,24 @@ public void seek() {
   @Override
   public void createTracks(ExtractorOutput extractorOutput, TrackIdGenerator idGenerator) {
     idGenerator.generateNewId();
-    this.output = extractorOutput.track(idGenerator.getTrackId(), C.TRACK_TYPE_TEXT);
-    output.format(Format.createImageSampleFormat(idGenerator.getFormatId(),
-        MimeTypes.APPLICATION_DVBSUBS, null, Format.NO_VALUE, initializationData, language, null));
+
+    TrackOutput output;
+    LanguageInfo language;
+
+    for (int i = 0; i < languages.size(); i++) {
+      language = languages.get(i);
+      idGenerator.generateNewId();
+
+      if (((language.programElementType & 0xF0 ) >> 4 ) == 2 ) {
+        language.languageCode += " for hard of hearing";
```

Comment:
@ojw28 , it was my first though but I did a quick hack until I had time to implement the needed flags for the audio and subtitle tracks